### ISSUE TITLE
`report-unused-eslint-disable-comments` not working

### ISSUE THREAD

**Issue by LvChengbin** [mantoni/eslint_d.js#146]
```sh
warning  Unused eslint-disable directive (no problems were reported from '@typescript-eslint/no-empty-function')
```
The warning message of `report-unused-elint-disable-comments` cannot be shown.

https://eslint.org/docs/user-guide/configuring#report-unused-eslint-disable-comments

**Comment by mantoni** [comment]
Does it work correctly when you invoke `eslint` directly? Also: Did you try `eslint_d restart`?

**Comment by LvChengbin** [comment]
@mantoni Yes, I tried  `eslint_d restart`. Using `eslint_d` can print other messages but no messages of `report-unused-eslint-disable-comments`.

This is a simple example:

```sh
% cat test.ts
const x = 1;
const y = 2;

console.log( x ); // eslint-disable-line


% yarn eslint test.ts

/test.ts
  2:7   error    'y' is assigned a value but never used                       @typescript-eslint/no-unused-vars
  4:19  warning  Unused eslint-disable directive (no problems were reported)

✖ 2 problems (1 error, 1 warning)


% yarn eslint_d test.ts

/test.ts
  2:7  error  'y' is assigned a value but never used  @typescript-eslint/no-unused-vars

✖ 1 problem (1 error, 0 warnings)
```

**Comment by mantoni** [comment]
Thank you for providing a reproducible use case. I'll look into it later this week. 🙏

**Comment by mantoni** [comment]
@LvChengbin I was able to reproduce the warning when I removed the blanks around the `x`:

```js
'use strict';

const x = 1;
const y = 2;

console.log(x); // eslint-disable-line
```

Regular eslint does not report a warning though, unless `--report-unused-disable-directives` is provided.

```
❯ node_modules/.bin/eslint bug.js

.../bug.js
  4:7  error  'y' is assigned a value but never used  no-unused-vars

✖ 1 problem (1 error, 0 warnings)
```

```
❯ node_modules/.bin/eslint bug.js --report-unused-disable-directives

.../bug.js
  4:7   error  'y' is assigned a value but never used                       no-unused-vars
  6:17  error  Unused eslint-disable directive (no problems were reported)

✖ 2 problems (2 errors, 0 warnings)
```

The behaviour of `eslint_d` is identical with `eslint` here. Am I holding it wrong?

**Comment by mantoni** [comment]
I was indeed holding it wrong. You're referring to the `reportUnusedDisableDirectives` config option, not the command line option. They behave differently.

I can now reproduce your issue.

**Comment by mantoni** [comment]
`v10.0.0` is out which uses the new `ESLint` API. From my testing this fixes your issue. Can you confirm?

**Comment by LvChengbin** [comment]
I upgraded my `eslint_d` to `v10.0.0` and it is working as expected.
Thank you so much. `eslint_d` helps me and saves me lots of time.
